client/lib/parsing.rs

```diff
@@ -0,0 +1,588 @@
+//! This module contains structs and helpers which are used by multiple subcommands related to
+//! creating deploys.
+
+use std::{convert::TryInto, fs, path::PathBuf, str::FromStr};
+
+use serde::{self, Deserialize};
+
+use casper_execution_engine::core::engine_state::executable_deploy_item::ExecutableDeployItem;
+use casper_node::{
+    crypto::{
+        asymmetric_key::{PublicKey as NodePublicKey, SecretKey},
+        hash::Digest,
+    },
+    types::{TimeDiff, Timestamp},
+};
+use casper_types::{
+    bytesrepr, CLType, CLValue, ContractHash, Key, NamedArg, RuntimeArgs, UIntParseError, URef,
+    U512,
+};
+
+use crate::{
+    cl_type,
+    deploy::DeployParams,
+    error::{Error, Result},
+    ExecutableDeployItemExt, TransferTarget,
+};
+
+pub(super) fn none_if_empty(value: &'_ str) -> Option<&'_ str> {
+    if value.is_empty() {
+        return None;
+    }
+    Some(value)
+}
+/// Handles providing the arg for and retrieval of the timestamp.
```

Comment:
I'm not sure there's any point to these small modules any more - they're pretty much all just called from one place each.  I'd be inclined to just move the parsing to code for the simpler ones to the callsites.  If you prefer to keep them here, I'm fine with that, but let's just drop the mods now.  So change `parsing::timestamp::parse()` to just something like `parsing::timestamp()`, etc.

Also, the comments are out of date now too.  They don't provide the arg now, they just parse it.

Reply:
Addressed this, thanks.